File: app/models/medium_detail.rb
class  MediumDetail < ApplicationRecord
  def actors
    self[:actors].split(/\s*,\s*/) || self[:actors]
  end

  def genre
    self[:genre].split(/\s*,\s*/) || self[:genre]
  end

  def released
    return self[:released] if self[:released] == "N/A"
    Date.strptime(self[:released], "%d %b %y") || self[:released]
  end
end

# == Schema Information
#
# Table name: medium_details
#
#  id            :integer          not null, primary key
#  title         :string
#  year          :string
#  rated         :string
#  released      :string
#  runtime       :string
#  genre         :string
#  director      :string
#  writer        :string
#  actors        :string
#  plot          :string
#  language      :string
#  country       :string
#  awards        :string
#  poster        :string
#  metascore     :string
#  imdb_rating   :string
#  imdb_votes    :string
#  imdb_id       :string
#  total_seasons :string
#  dvd           :string
#  box_office    :string
#  production    :string
#  website       :string
#  created_at    :datetime         not null
#  updated_at    :datetime         not null
#

File: omdb-react/src/app/MediumDetail.js
import React from 'react';

const MediumDetail = ({details}) => {
  const {title, runtime, released, actors, plot, imdb_rating} = details;
  return(
    <>
      <h1 className="medium-details title">{title}</h1>
      <p>{plot}</p>
      <table>
      <thead>
        <tr>
            <th colspan="2">Stats</th>
        </tr>
    </thead>
        <tr>
          <th>IMDB Rating</th>
          <td>{imdb_rating}</td>
        </tr>
        <tr>
          <th>Date Released</th>
          <td>{released}</td>
        </tr>
        <tr>
          <th>Runtime</th>
          <td>{runtime}</td>
        </tr>
      </table>
      <h2>Actors:</h2>
      <ul className="medium-details actors-list">{
        actors.map(
          (actor) => {
            return <li>{actor}</li>
          }
        )}
      </ul>
    </>
  );
}

export default MediumDetail;